src/internal/passwordHash.ts

```diff
@@ -0,0 +1,13 @@
+import * as Scrypt from "@noble/hashes/scrypt"
+import * as Data from "effect/Data"
+import * as Effect from "effect/Effect"
+
+class OptionsError extends Data.TaggedError("OptionsError")<{
+  cause: unknown
+}> {}
```

Comment:
Let's also put the `options` here? I'd probably not include the `input` and `salt` out of an abundance of caution to not accidentally expose any of those).

Reply:
It has pretty good error messages mentioning exactly what was wrong. Should we not focus in exposing these ones?

Reply:
Yeah we are exposing those through the `cause`. Do you think `options` wouldn't add any value? If that's the case then I'm happy to not include it.

Reply:
Don't think so. It's not like you pass in different value per user.

Reply:
To be honest, you shouldn't even use scrypt :D You should use argon2 - should we add that to the docs? basically https://www.latacora.com/blog/2018/04/03/cryptographic-right-answers/

Reply:
Yeah that makes sense